### PR TITLE
Add string constraints

### PR DESCRIPTION
**Pull request by Pyrofab**
This PR implements the length bound constraints that had been previously planned, as well as a regular expression pattern constraint replacing the less versatile prefix and suffix constraints.
I also took the opportunity to add `Target` information to every `Setting` annotation.

### CODE REVIEW

File: src/main/java/me/zeroeightsix/fiber/annotation/Setting.java [hunk 1]
```diff
@@ -1,33 +1,58 @@
...
+        @Retention(RetentionPolicy.RUNTIME)
+        @interface MinStringLength {
+            int value();
+        }
+
+        @Target(ElementType.FIELD)
+        @Retention(RetentionPolicy.RUNTIME)
+        @interface MaxStringLength {
+            int value();
+        }
```
Comment (zeroeightysix) [inline src/main/java/me/zeroeightsix/fiber/annotation/Setting.java]: While I'm not exactly _against_ the naming of these, they could be shorter. We're dealing with already very long, verbose names (`Setting.Constrain.MaxStringLength`).
Basically, anything dealing with constraining length is *probably* a string, so can I propose renaming this to `MinLength` and `MaxLength` for brevity?

File: src/main/java/me/zeroeightsix/fiber/builder/constraint/ConstraintsBuilder.java [hunk 2]
```diff
@@ -30,9 +32,23 @@ public ConstraintsBuilder(List<Constraint<? super T>> sourceConstraints, Class<T
...
 
+	public ConstraintsBuilder<T> minStringLength(int min) {
+		addStringLengthLowerBound(min);
+		return this;
+	}
+
+	public ConstraintsBuilder<T> maxStringLength(int min) {
+		addStringLengthUpperBound(min);
+		return this;
+	}
```
Comment (zeroeightysix) [inline src/main/java/me/zeroeightsix/fiber/builder/constraint/ConstraintsBuilder.java]: Can these methods also be implemented for the composite constraint builder?

(+ the same naming suggestion applies to these)

File: src/main/java/me/zeroeightsix/fiber/annotation/Setting.java (hunk 1 above)
Comment (Pyrofab) [reply src/main/java/me/zeroeightsix/fiber/annotation/Setting.java]: Wouldn't it imply that this annotation can work on collections too ?

Comment (zeroeightysix) [reply src/main/java/me/zeroeightsix/fiber/annotation/Setting.java]: You're right. Good point.

**Comment by zeroeightysix** [comment]
Also please bump version to `0.9.0` (in gradle.properties)

**Comment by zeroeightysix** [comment]
LGTM - Ready to merge?

**Comment by zeroeightysix** [comment]
Thanks!

**Pull request merged**